app.js

```diff
@@ -64,23 +54,48 @@ app.post('/api/fcc-comment/sheets', routes.fccCommentSheets);
 
 app.use(reactRouted);
 app.use(express.static(__dirname + '/public', {maxAge: 3600000}));
-app.use(function(req, res, next) {
+
+/**
+ * We need to make sure that resources are presented to the
+ * user in the appropriate locale, so any requests without
+ * a locale should first be locale-enriched based on the
+ * request headers we receive from the client.
+ */
+function routeBasedOnLocale(req, res, next) {
   var location = url.parse(req.url).pathname;
   var search = url.parse(req.url).search || "";
+
   // Get a valid locale from the path and header
   var parsed = locationParser(req.headers["accept-language"], location);
   var parsedLocale = parsed.locale;
   var parsedRedirect = parsed.redirect;
+
   // See if we should redirect.
   if (parsedRedirect) {
-    res.redirect(301, "/" + parsedLocale + parsedRedirect + search);
+    let newUrl = "/" + parsedLocale + parsedRedirect + search;
+    if (newUrl === req.url) {
+      console.warn("Received meaningless redirect: new URL is identical to original URL. Skipping to next()");
```

Comment:
This is interesting. I would like to know the cases where we had meaningless redirects.